File: Python/hashTable/HashTable.py

import sys

class HashTable:
    
    def __init__(self, table_size):
        if table_size < 1:
            print('Error: table size must be positive number!')
            sys.exit(1)
        self.table_size = table_size
        self.table = [[] for i in range(table_size)]
    
    def hash_func(self, key):
        return key % self.table_size

    def insert(self, key):
        self.table[self.hash_func(key)].append(key)
    
    def show(self):
        for linked_list in self.table:
            if linked_list:
                for key in linked_list:
                    print('%d' % key, end='')
                print('')

    def search(self, key):
        if key in self.table[self.hash_func(key)]:
            return True
        return False

File: java/src/main/java/org/example/hashTable/HashTable.java
package org.example.hashTable;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import java.util.stream.IntStream;

public class HashTable {

    private final int tableSize;
    private final List<List<Integer>> table = new ArrayList<>();

    public HashTable(int tableSize) {
        if (tableSize < 1) {
            System.out.println("Error: table size must be positive number!");
            System.exit(1);
        }
        this.tableSize = tableSize;
        IntStream.rangeClosed(0, tableSize)
                .boxed()
                .forEach(index -> table.add(index, new ArrayList<>()));
    }

    private int hashFunc(int key) {
        return key % tableSize;
    }

    public void insert(int key) {
        final var hashKey = hashFunc(key);
        table.get(hashKey).add(key);
    }

    public boolean search(int key) {
        final var hashKey = hashFunc(key);
        final var tableByKey = table.get(hashKey);
        return tableByKey.stream().anyMatch(integer -> integer == key);
    }

    public void show() {
        table.stream()
                .filter(Objects::nonNull)
                .forEach(internalList -> {
                    internalList.forEach(key -> System.out.print(key + " "));
                    System.out.print("\n");
                });
    }

    public List<List<Integer>> getTable() {
        return table;
    }

}
